channels/audin/client/audin_main.c

```diff
@@ -176,7 +176,14 @@ static int audin_process_formats(IWTSVirtualChannelCallback* pChannelCallback, w
 			/* Store the agreed format in the corresponding index */
 			callback->formats[callback->formats_count++] = format;
 			/* Put the format to output buffer */
-			Stream_EnsureRemainingCapacity(out, 18 + format.cbSize);
+			if (!Stream_EnsureRemainingCapacity(out, 18 + format.cbSize))
+			{
+				free(callback->formats);
+				Stream_Free(out, TRUE);
+				return 1;
```

Comment:
This is not related to your patch, but is it me or this function returns 1 in case of error ? (I'm starting to think that nobody cares about the result...)